File: scripts/algorithms/Valid Perfect Square/Valid Perfect Square.js
var isPerfectSquare = function(num) {
    let low = 1;
    let high = 100000;
    while(low <= high){
        let mid = (low + high) >> 1;
        let sqrt = mid * mid
        if( sqrt == num) {
            return true;
        
        }else if(num > sqrt ){
            low = mid + 1
        }else {
            high = mid - 1
        }
    }
    return false;
};

File: scripts/algorithms/V/Valid Perfect Square/Valid Perfect Square.java
// Runtime: 0 ms (Top 100.00%) | Memory: 40.6 MB (Top 74.32%)

class Solution {
    public boolean isPerfectSquare(int num) {
        long start = 1;
        long end = num;

        while(start<=end){
            long mid = start +(end - start)/2;

            if(mid*mid==num){
                return true;
            }
            else if(mid*mid<num){
                start = mid+1;
            }
            else
                end = mid-1;
        }
        return false;
    }
}